### PR TITLE
tabula-java: Init at 1.04

### PR DESCRIPTION
**Pull request by jbaum98**
###### Motivation for this change

Add the CLI tool for `tabula`, `tabula-java`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/tabula-java/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  meta = with stdenv.lib; {
+    description = "A library for extracting tables from PDF files.";
+    longDescription = ''
+      tabula-java is the table extraction engine that powers
+      Tabula. You can use tabula-java as a command-line tool to
+      programmatically extract tables from PDFs.
+    '';
+    homepage = "https://tabula.technology/";
+    license = licenses.mit;
+    maintainers = [ maintainers.jbaum98 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tabula-java/default.nix:33]: Please put this into a separate commit with the message ``maintainers: add jbaum98``.

File: pkgs/applications/misc/tabula-java/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  phases = [ "installPhase" ];
+
+  installPhase = ''
+    mkdir -pv $out/share/tabula-java
+    cp -v $src $out/share/tabula-java/tabula-java.jar
+
+    makeWrapper ${jre}/bin/java $out/bin/tabula-java --add-flags "-jar $out/share/tabula-java/tabula-java.jar"
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tabula-java/default.nix:24]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/misc/tabula-java/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  version = "1.0.4";
+
+  src = fetchurl {
+    url = "https://github.com/tabulapdf/tabula-java/releases/download/v${version}/tabula-${version}-jar-with-dependencies.jar";
+    sha256 = "12d0jcc3j0q8jbqv0gzfiif7c8ig37834vb8yq0jnyr0s72k30xw";
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  phases = [ "installPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tabula-java/default.nix:15]:
```suggestion
```

Please disable the phases you don't want to run eg dontUnpack = true;

File: pkgs/applications/misc/tabula-java/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, jre, makeWrapper }:
+
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tabula-java/default.nix:4]:
```suggestion
{ stdenv, fetchurl, jre, makeWrapper }:

stdenv.mkDerivation rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/tabula-java/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/tabula-java/default.nix:33]: Sorry about that. I've posted the default message but that did not apply here. This change should go into the package init commit but you need to add yourself as a maintainer first which should go into its own commit.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by jbaum98** [comment]
I was confused because I'm already in the maintainers file [here](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L4088), but I used the wrong name. I'll fix that.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110905` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>tabula-java</li>
</ul>
</details>
